hibernate-core/src/main/java/org/hibernate/cfg/BytecodeSettings.java

```diff
@@ -20,7 +20,10 @@ public interface BytecodeSettings {
 	 * At present only bytebuddy is supported, bytebuddy being the default since version 5.3.
 	 *
 	 * @settingDefault {@code "bytebuddy"}
+	 * @deprecated Will be removed, users can override the default provider by registering a
+	 * service for the BytecodeProvider interface. See HHH-17643
```

Comment:
Great that you remembered to update the documentation :+1: 

However I'd change the language a little: people don't "register"  anything to t he service loader, but things will be found automatically as long as they have the right dependency on the classpath.

So it might be more useful to explain something among those lines: "Hibernate ORM will use the BytecodeProvider implementation it finds on the classpath loading it via the standard ServiceLoader mechanism. Currently there is only a single implementation, which is included in Hibernate ORM so it's currently not possible to override this."

Reply:
Changed the description with this, thanks.